modules/core/04-channel/keeper/packet.go

```diff
@@ -287,11 +287,8 @@ func (k Keeper) WriteAcknowledgement(
 		return errorsmod.Wrap(types.ErrChannelNotFound, packet.GetDestChannel())
 	}
 
-	if !channel.IsOpen() {
-		return errorsmod.Wrapf(
-			types.ErrInvalidChannelState,
-			"channel state is not OPEN (got %s)", channel.State.String(),
-		)
+	if !slices.Contains([]types.State{types.OPEN, types.FLUSHING}, channel.State) {
```

Comment:
This should contained `FLUSHCOMPLETE` as well? If we finish flushing our packet sends, the counterparty might not have finished flushing theirs (we still have packets to receive)

Reply:
we would also need to add `FLUSHCOMPLETE` to recvPacket then here:

https://github.com/cosmos/ibc-go/blob/main/modules/core/04-channel/keeper/packet.go#L121

Reply:
`FLUSHCOMPLETE` says we are finished processing outbound packets sent from our side, but we must be able to support processing (recving) packets which were sent from the other channel end. 

So we should add `FLUSHCOMPLETE` to recvPacket and writeAck, but acknowledgePacket should only work in `OPEN` and `FLUSHING`.

Reply:
Yes, that's right. Good catch!

Reply:
Updated. I will look into having an e2e for this.